tests/esm3/test_esm3_embeddings.py

```
@@ -0,0 +1,235 @@
+    with pytest.raises(ValueError, match="Unsupported ESM-C model"):
+        ESMCCore(config={"model_name": "nope"})
+
+    core = ESMCCore(config={"device": "cpu"})
+    with pytest.raises(ValueError, match="model_name"):
+        core.embed("ACD", options={"model_name": "esmc_600m"})
+
+
+def test_esm3_core_hidden_states_are_rejected(fake_esm_sdk: type[_FakeSDKModel]) -> None:
+    from boileroom.models.esm3.core import ESM3Core
+
+    with pytest.raises(ValueError, match="hidden_states.*ESM3"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)